### PR TITLE
[AD-980] Handle in-driver date add calculation for literals.

### PR DESCRIPTION
**Pull request by birschick-bq**
### Summary

[AD-980] Handle in-driver date add calculation for literals.

### Description

Perform in-driver date addition for date literals, include CURRENT_DATE.

### Related Issue

https://bitquill.atlassian.net/browse/AD-980

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq

### CODE REVIEW

File: src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java [hunk 1]
```diff
@@ -757,6 +759,7 @@ private static Operand reformatObjectIdLiteral(
 
     private static class DateFunctionTranslator {
 
+        private static final String CURRENT_DATE = "CURRENT_DATE";
```
Comment (andiemontoyeah) [inline src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java:762]: Check if we can also support CURRENT_TIMESTAMP

File: src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java [hunk 2]
```diff
@@ -786,9 +789,68 @@ private static Operand translateCurrentTimestamp(final Instant currentTime) {
...
+            final BsonDocument document0 =  BsonDocument.parse("{field: " + queryValue0 + "}");
+            final BsonDocument document1 =  BsonDocument.parse("{field: " + queryValue1 + "}");
+
+            long sum = 0L;
+            for (BsonValue v : new BsonValue[]{document0.get("field"), document1.get("field")}) {
+                switch (v.getBsonType()) {
+                    case DOUBLE:
+                        sum += v.asDouble().longValue();
+                        break;
+                    case STRING:
```
Comment (andiemontoyeah) [inline src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java]: Can these cases occur? Do we need unit tests for them? I think it caused a slightest of drops in branch coverage

File: src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java [hunk 3]
```diff
@@ -798,6 +798,32 @@ void testWhereTimestampAdd() throws SQLException {
                 operations.get(3));
     }
 
+    @Test
+    @DisplayName("Tests TIMESTAMPADD on right in WHERE clause.")
+    void testWhereTimestampAddOnRight() throws SQLException {
```
Comment (andiemontoyeah) [inline src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java]: can we also add test for TIMESTAMPADD on left?

File: src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java [hunk 4]
```diff
@@ -786,9 +789,68 @@ private static Operand translateCurrentTimestamp(final Instant currentTime) {
...
+            // Otherwise, perform addition on server.
             return new Operand("{ \"$add\":" + "[" + Util.commaList(strings) + "]}");
         }
 
+        private static boolean isDateLiteralCandidate(final RexCall call, final List<Operand> strings) {
+            final boolean allLiterals = call.getOperands().stream()
+                    .allMatch(op -> {
+                        final SqlKind opKind = op.getKind();
+                        final String opName = op.toString();
+                        return opKind == SqlKind.LITERAL
```
Comment (andiemontoyeah) [inline src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java:809]: can we add unit tests operand is date or timestamp literal?

**Review by andiemontoyeah — COMMENTED** [review]
some comments but lgtm!

File: src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java (hunk 1 above)
Comment (birschick-bq) [reply src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java:762]: OK.

File: src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java (hunk 3 above)
Comment (birschick-bq) [reply src/test/java/software/amazon/documentdb/jdbc/query/DocumentDbQueryMappingServiceDateTimeTest.java]: OK

File: src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java (hunk 4 above)
Comment (birschick-bq) [reply src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java:809]: OK.

File: src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java (hunk 2 above)
Comment (birschick-bq) [reply src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java]: Removed data types that won't be found.

**Review by andiemontoyeah — APPROVED** [review]
No additional comments from me 👍

**Review by affonsov — APPROVED** [review]

**Pull request merged**